packages/myst-cli/src/build/html/index.ts

```diff
@@ -81,22 +81,40 @@ function rewriteAssetsFolder(directory: string, baseurl?: string): void {
 }
 
 /**
- * Build a MyST project as a static HTML deployment
+ * Get the baseurl from BASE_URL or common deployment environments
  *
  * @param session session with logging
- * @param opts configuration options
  */
-export async function buildHtml(session: ISession, opts: any) {
-  const template = await getMystTemplate(session, opts);
-  // The BASE_URL env variable allows for mounting the site in a folder, e.g., github pages
-  const baseurl = process.env.BASE_URL;
+function get_baseurl(session: ISession): string | undefined {
+  let baseurl;
+  // BASE_URL always takes precedence. If it's not defined, check common deployment environments.
+  if ((baseurl = process.env.BASE_URL)) {
+    session.log.info('BASE_URL environment overwrite is set');
+  } else if ((baseurl = process.env.READTHEDOCS_CANONICAL_URL)) {
+    session.log.info('Building inside a ReadTheDocs environment');
```

Comment:
It looks like this is the full URL, and not just a part of it? (based on [docs here](https://docs.readthedocs.io/en/stable/reference/environment-variables.html#envvar-READTHEDOCS_CANONICAL_URL)).

I think that we should validate and only be taking the path part of the URL. in this case something like `/en/latest`. Perhaps something like:

```suggestion
  } else if (process.env.READTHEDOCS_CANONICAL_URL) {
    baseurl = new URL(process.env.READTHEDOCS_CANONICAL_URL).pathname.replace(/\/$/, '');
    session.log.info(`Building inside a ReadTheDocs environment with BASE_URL="${baseurl}"`);
```

> The base URL is absolute and should not end with a trailing slash.

Now that there is a function, we could test the URL against that even a simple regexp like this: `/^\/(.*)([^\/])$/` and raising an error if it doesn't match.

Reply:
Kay, I've added the sanitization, with a bit different info reporting.

I was looking at adding a unit test, but I couldn't find a framework to use, and I'm not sure of how/where `vitest` are defined